ray_on_golem/server/services/golem/golem.py

```diff
@@ -272,14 +280,17 @@ def _score_with_provider_data(
         return 0.5 + (0.5 * (provider_pos / len(prescored_providers)))
 
     @staticmethod
-    async def _get_provider_desc(context: WorkContext):
-        return f"{await context.get_provider_name()} ({await context.get_provider_id()})"
+    async def get_provider_desc(activity: Activity):
+        return (
+            f"{await activity.agreement.proposal.get_provider_name()}"
+            f"({await activity.agreement.proposal.get_provider_id()})"
+        )
```

Comment:
this would look nicer, but of course it's optional :)

```suggestion
    async def get_provider_desc(activity: Activity):
        proposal = activity.agreement.proposal
        return (f"{await proposal.get_provider_name()} ({await proposal.get_provider_id()})")
```

Reply:
but on the other hand, my question is why are we eliminating the call from the context?

Reply:
> why are we eliminating the call from the context

I don't understand. I needed it to work on activity level so I decided there is no point of keeping both methods here. Context is still capable to return those values

Reply:
```suggestion
    async def get_provider_desc(activity: Activity):
        proposal = activity.agreement.proposal
        provider_name = await proposal.get_provider_name()
        provider_id = await proposal.get_provider_id()
        return f"{provider_name} ({provider_id})"
```
My "nicer" take would look like this... 😅

Reply:
That is to much for me and the blue's suggestions fits in a single line